controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -152,6 +159,45 @@ func (r *OCSInitializationReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	if _, ok := r.availableCrds[ClusterClaimCrdName]; ok {
+		crdHandler := handler.TypedEnqueueRequestsFromMapFunc(func(_ context.Context, obj client.Object) []reconcile.Request {
+			if obj.GetName() == ClusterClaimCrdName {
+				return []reconcile.Request{
+					{
+						NamespacedName: initNamespacedName,
+					},
+				}
+			}
+			return []reconcile.Request{}
+		})
+
+		crdPredicate := predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				if metaObj, ok := e.Object.(metav1.Object); ok {
+					return metaObj.GetName() == ClusterClaimCrdName
+				}
+				return false
+			},
+			DeleteFunc: func(_ event.DeleteEvent) bool {
+				return false
+			},
+			UpdateFunc: func(_ event.UpdateEvent) bool {
+				return false
+			},
+			GenericFunc: func(_ event.GenericEvent) bool {
+				return false
+			},
+		}
+		if err := r.controller.Watch(source.Kind[client.Object](r.cluster.GetCache(),
+			&apiextensionsv1.CustomResourceDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ClusterClaimCrdName,
+				},
+			}, crdHandler, crdPredicate)); err != nil {
+			return reconcile.Result{}, fmt.Errorf("unable to watch CRD")
+		}
```

Comment:
@umangachapagain This will not achieve the desired result. You cannot just add watches to a controller after the manager started, the cache will become incoherent and inconsistent. The only way to do it within the process is to close all controllers stop and reset the manager then start the manager again. What you are doing here can only work if the client we use is cacheless, but then the informers will not work properly and there will be a runtime penalty for lists and gets